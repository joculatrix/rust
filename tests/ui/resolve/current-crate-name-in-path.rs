//! Ensure E0433 suggests replacing the crate's name with `crate` when the
//! current crate's name is used in a path.

mod bar {
    pub fn baz() {}
}

use current_crate_name_in_path::bar::baz;
//~^ ERROR failed to resolve: use of unresolved module or unlinked crate `current_crate_name_in_path`
//~| SUGGESTION crate

fn main() {}
