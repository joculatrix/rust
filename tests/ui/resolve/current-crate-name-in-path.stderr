error[E0433]: failed to resolve: use of unresolved module or unlinked crate `current_crate_name_in_path`
  --> $DIR/current-crate-name-in-path.rs:8:5
   |
LL | use current_crate_name_in_path::bar::baz;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `current_crate_name_in_path`
   |
help: the current crate name can not be used in a path, use the keyword `crate` instead
   |
LL - use current_crate_name_in_path::bar::baz;
LL + use crate::bar::baz;
   |

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0433`.
